sklearn/metrics/cluster/tests/test_common.py

```diff
@@ -183,3 +184,39 @@ def generate_formats(y):
         y_true_gen = generate_formats(y_true)
         for (y_true_fmt, fmt_name) in y_true_gen:
             assert score_1 == metric(X, y_true_fmt)
+
+
+@pytest.mark.parametrize(
+    "metric_name",
+    [name for name in SUPERVISED_METRICS]
+)
+# only the supervised metrics support single sample
+def test_single_sample(metric_name):
+    metric = SUPERVISED_METRICS[metric_name]
+    for i, j in product([0, 1], repeat=2):
+        metric([i], [j])
+
+
+@pytest.mark.parametrize(
+    "metric_name",
+    [name for name in dict(SUPERVISED_METRICS, **UNSUPERVISED_METRICS)]
+)
+def test_inf_nan_input(metric_name):
+    if metric_name in SUPERVISED_METRICS:
+        invalids = [([0, 1], [np.inf, np.inf]),
+                    ([0, 1], [np.nan, np.nan]),
+                    ([0, 1], [np.nan, np.inf])]
+
+        metric = SUPERVISED_METRICS[metric_name]
+        with pytest.raises(ValueError, match="contains NaN, infinity"):
+            for y_true, y_score in invalids:
+                metric(y_true, y_score)
+    else:
+        X = np.random.randint(10, size=(2, 10))
+        invalids = [(X, [np.inf, np.inf]),
+                    (X, [np.nan, np.nan]),
+                    (X, [np.nan, np.inf])]
+        metric = UNSUPERVISED_METRICS[metric_name]
```

Comment:
Everything below this line is strictly identical in both `if` branches. Only `invalids` and metric needs to be defined there I think?

Or rather,
```py
for args in invalids:
    metric(*args)
```
as it's confusing to see `metric(y_true, y_score)` for unsupervised ones.